crates/bitwarden-core/src/platform/client_platform.rs

```diff
@@ -5,15 +5,54 @@ use super::{
 };
 use crate::{error::Result, Client};
 
+/// A struct containing platform utilities.
 pub struct ClientPlatform<'a> {
     pub(crate) client: &'a Client,
 }
 
 impl<'a> ClientPlatform<'a> {
+    /// Will generate a fingerprint based on the `input`. Given the same `input` This
+    /// method will result in the exact same output.
+    ///
+    /// # Examples
+    /// ```rust
+    /// use bitwarden_core::{Client, platform::FingerprintRequest};
+    ///
+    /// async fn test() {
+    ///     let client = Client::test_account().await;
+    ///     let fingerprint_response = client.platform()
+    ///         .fingerprint(&FingerprintRequest {
+    ///             fingerprint_material: "my_material".to_owned(),
+    ///             public_key: "...public key...".to_owned(),
```

Comment:
question/suggestion: can these fields take any values, or do they expect the fields to be formatted in a certain way, e.g. base64? If yes, could we let the example clarify that somehow?

Reply:
Just realized that these docs are tested and so this must work 🙃

Reply:
Well, the test runs the code at the top level, which means it sees the `test` function but doesn't actually call it.

If you want your test code to be called, it either needs to be outside of a function, or you need to call the function at the top level of the code block.

With sync functions you can just call them:
```rust
    /// Does a thing
    ///
    /// ```rust
    /// fn test() {
    ///     panic!("This test now fails")
    ///     Ok(())
    /// }
    /// # test();
    /// ```
```

With async functions for example you can do this:
```rust
    /// Does a thing
    ///
    /// ```rust
    /// async fn test() {
    ///     panic!("This test now fails")
    ///     Ok(())
    /// }
    /// # tokio_test::block_on(test());
    /// ```
```

Note the last line: `/// # tokio_test::block_on(test());` which runs the async test using `tokio_test`. It's at the top level of the code block, and not inside a function, so it gets run. The `#` also hides the line from the generated doc, while still running it during tests, which makes the docs nicer to look at:

![image](https://github.com/user-attachments/assets/19f4dff8-526a-44ad-ade7-1cfaa7e8bc69)

Reply:
I did _not_ notice that the code was inside of functions, good catch!